mesa/visualization/AgentPortrayalStyle.py

```diff
@@ -0,0 +1,28 @@
+from dataclasses import dataclass
+
+
+@dataclass
+class AgentPortrayalStyle:
+    """Dataclass to define agent portrayal styles in a proper structured way."""
+
+    color: str = "red"
+    marker: str = "o"
+    size: int = 10
+    zorder: int = 1
+    alpha: float = 1.0  # Supports transparency
+    linewidths: float = 1.0  # line width for markers
+    edgecolors: str = "black"  # edge color for markers
+    loc: tuple[float, float] | None = None  # stores agent position
+
+    def to_dict(self):
```

Comment:
I would suggest doing it the other way arround. In get_agent_data if the return is a dict cast it to a AgentPortrayalStyle and issue a warning